### PR TITLE
TYPO3 12 compatibility

### PR DESCRIPTION
**Pull request by kitzberger**
Refactoring to achieve compatibility for TYPO3 12.4 and EXT:powermail 12.x

### CODE REVIEW

File: Classes/ViewHelpers/FormViewHelper.php [hunk 1]
```diff
@@ -268,7 +274,9 @@ protected function renderTrustedPropertiesField()
      */
     protected function renderFormUidField()
     {
-        $this->addHiddenField($this->prefixFieldName('mail[form]'), $this->arguments['formUid']);
+        $fieldName = $this->prefixFieldName('mail[form]');
+        $this->registerFieldNameForFormTokenGeneration($fieldName);
+        $this->addHiddenField($this->prefixFieldName($fieldName), $this->arguments['formUid']);
```
Comment (nabossha) [inline Classes/ViewHelpers/FormViewHelper.php:279]: I have tested your refactoring on a recently to LTS12 upgraded project and found `$this-prefixFieldName` here seems "doubled" and produces `tx_powermail_pi1[tx_powermail_pi1][mail][form]` but it should be `tx_powermail_pi1[mail][form]`

Comment (kitzberger) [reply Classes/ViewHelpers/FormViewHelper.php]: @nabossha, good catch! Thanks for pointing out. It's fixed now :muscle:

**Review by nabossha — APPROVED** [review]
works for me thanks!

**Comment by lukaszuznanski** [comment]
Merging into master, let's work on other PRs for other fixes related to v12. I will wait some time before official release.

**Pull request merged**